### PR TITLE
Fix: Adding favicon to Kedro Demo

### PR DESCRIPTION
**Pull request by jitu5**
## Description

Resolves [#1471](https://github.com/kedro-org/kedro-viz/issues/1471)

Added favicon on the demo website: https://demo.kedro.org/

## Checklist

- [X] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [x] Added new entries to the `RELEASE.md` file
- [x] Added tests to cover my changes

<a href="https://gitpod.io/#https://github.com/kedro-org/kedro-viz/pull/1509"></a>

### CODE REVIEW

**Comment by jitu5** [comment]
@rashidakanchwala @ravi-kumar-pilla  Below approach also works by creating separate end point for favicon. Please let me know which one fits as per our best practices. 

```
 # Serve the favicon.ico file from the "html" directory
 @app.get('/favicon.ico', include_in_schema=False)
 async def favicon():
     return FileResponse(_HTML_DIR / "favicon.ico")
```

**Comment by rashidakanchwala** [comment]
Actually, I like both these approaches. 

I did find the second approach of creating an endpoint as the most common way of serving favicon on FastAPi when I did some research yesterday.

**Review by ravi-kumar-pilla — APPROVED** [review]
Tested on GitPod and looks good to me. Can you please update the release doc before merging ? Thank you

File: RELEASE.md [hunk 1]
```diff
@@ -15,8 +15,10 @@ Please follow the established format:
 ## Bug fixes and other changes
 
 - Fix to search for a '<lambda' Python function in the sidebar. (#1497)
+- Favicon added to Kedro-viz Demo site. (#1509)
```
Comment (rashidakanchwala) [inline RELEASE.md:18]: we use a more imperative tone

```suggestion
- Add favicon to Kedro-Viz. (#1509)
```

File: package/tests/test_api/test_apps.py [hunk 2]
```diff
@@ -62,3 +62,18 @@ def test_reload_endpoint_return_304_when_content_has_not_changed(
         # when the etag has changed, the server will return a 200
         response = client.get("/api/reload", headers={"If-None-Match": "new etag"})
         assert response.status_code == 200
+
+
+class TestFaviconEndpoint:
+    @pytest.fixture
```
Comment (rashidakanchwala) [inline package/tests/test_api/test_apps.py:68]: We don't need this pytest.fixture. We can use client which is defined in conftest.py and just do this directly 

```
 def test_favicon_endpoint(self, client):
        response = client.get("/favicon.ico")
        assert response.status_code == 200
        assert response.headers["content-type"] in [
            "image/x-icon",
            "image/vnd.microsoft.icon",
        ]
```

Look at line 11,12 in the same file. And let me know if you have more questions.

File: package/tests/test_api/test_apps.py [hunk 3]
```diff
@@ -62,3 +62,18 @@ def test_reload_endpoint_return_304_when_content_has_not_changed(
...
+    def test_client(self):
+        app = apps.create_api_app_from_project(mock.MagicMock(), autoreload=True)
+        return TestClient(app)
+
+    def test_favicon_endpoint(self, test_client):
+        response = test_client.get("/favicon.ico")
+        assert response.status_code == 200
+        assert response.headers["content-type"] in [
+            "image/x-icon",
+            "image/vnd.microsoft.icon",
```
Comment (rashidakanchwala) [inline package/tests/test_api/test_apps.py:78]: QQ - we have 2 content-types here, is it because of linux and windows tests?

Comment (jitu5) [reply package/tests/test_api/test_apps.py:78]: Here 'image/x-icon' is  commonly used MIME type for ICO files, some server configurations might use 'image/vnd.microsoft.icon' as ICO files content type, in fact our test setup uses 'image/vnd.microsoft.icon' as ICO files content type. So added both content type for checks.

**Review by rashidakanchwala — APPROVED** [review]
LGTM! and congrats on your first python PR!!

**Pull request merged**